[WIP] Add uk validation

deBhal opened the pull request:
This PR adds validation for the uk tld form, using a JSON Schema added to Calypso (for now).

![checkout_ _julesauspremiumtest_ _wordpress_com](https://user-images.githubusercontent.com/5952255/34709257-35444884-f562-11e7-9ef1-6c6a618115e7.jpg)

Stacked on https://github.com/Automattic/wp-calypso/pull/20750

a8ci18n commented on client/components/domains/registrant-extra-info/uk-form.jsx:
```diff
@@ -137,12 +164,31 @@ class RegistrantExtraInfoUkForm extends React.PureComponent {
+	errorMessageFromCode( errorCode ) {
+		return (
+			this.errorMessages[ errorCode ] ||
+			this.props.translate( 'There was a problem with this field.' )
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'There was a problem with this field' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

deBhal commented:
I have some unresolved questions around using the endpoint (how long do we want to keep the schema around? what should we do if there's a network error?), so I've split that off and added the schema to calypso for now.

I started refactoring fr to use the new improvements, but that got noisy very quickly, so I'll punt that to a separate PR too.
